CDU001-impl

fladson opened the pull request:
Pull request referente a tarefa final.

fladson committed:
Criada classes (BO, Jpa Controller, Bean) referentes ao Estagiario #14

Não foi feito nenhuma alteração quanto as páginas JSF, pois ainda estou
configurando o postgres localmente.

Favor analisar pois não foi testado a inserção nem a edição da entidade.

fladson committed:
Implementado pagina JSF e metodo incluir para entidade Estagiario

Não foi realizado os testes desta implementação por motivos de
configuração local do servidor.

tacianosilva commented:
@fladson segue os poblemas encontrados durante os testes da implementação do caso de uso:

Problema 1:
javax.servlet.ServletException: Cannot find component with identifier "messages" referenced from "j_idt19:j_idt20".
    javax.faces.webapp.FacesServlet.service(FacesServlet.java:606)
Detalhes 1:
Ele procura o componente messages no form do botão, contudo o componente está noutro form. Isso não deveria ocorrer.
Solução 1a:
Adicionei a tag <p:messages id="messages" /> no início do segundo form.

tacianosilva commented:
@fladson Problema 2: java.lang.IllegalArgumentException: An instance of a null PK has been incorrectly provided for this find operation.
Solução: Na anotação @SequenceGenerator(name="ESTAGIARIO_CODIGO_GENERATOR", sequenceName="ESTAGIARIO_CODIGO_SEQ") basta acrescentar allocationSize = 1. A anotação fica assim: @SequenceGenerator(name="ESTAGIARIO_CODIGO_GENERATOR", sequenceName="ESTAGIARIO_CODIGO_SEQ", allocationSize = 1)

tacianosilva commented:
@fladson não há formulário implementado para incluir um estagiário?

tacianosilva committed:
As modificações foram feitas para remover dois pequenos problemas
identificados e descritos abaixo. Também não há formulário implementado
para incluir um estagiário. Issue #14 e Issue #19

Problema 1:
javax.servlet.ServletException: Cannot find component with identifier
"messages" referenced from "j_idt19:j_idt20".
javax.faces.webapp.FacesServlet.service(FacesServlet.java:606)
Detalhes 1:
Ele procura o componente messages no form do botão, contudo o componente
está noutro form. Isso não deveria ocorrer.
Solução 1a:
Adicionei a tag no início do segundo form.

Problema 2: java.lang.IllegalArgumentException: An instance of a null PK
has been incorrectly provided for this find operation.
Solução: Na anotação
@SequenceGenerator(name="ESTAGIARIO_CODIGO_GENERATOR",
sequenceName="ESTAGIARIO_CODIGO_SEQ") basta acrescentar allocationSize =
1. A anotação fica assim:
@SequenceGenerator(name="ESTAGIARIO_CODIGO_GENERATOR",
sequenceName="ESTAGIARIO_CODIGO_SEQ", allocationSize = 1)

fladson commented:
Irei providenciar a questão do formulário.

fladson committed:
Formulário de cadastro de estagiário

Peço desculpas pelo desatento, estava commitado e acabei esquecendo de
enviar no último push, pois ainda estava alterando algumas coisas.

fladson committed:
Retirando formulário antigo incompleto(lixo)
